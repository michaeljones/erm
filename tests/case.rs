extern crate erm;
extern crate im;
extern crate insta;
extern crate logos;
extern crate unindent;

mod common;

mod case {

    use common::eval;

    #[test]
    fn case_statement() {
        let src = r#"
        module Main exposing (..)

        toText arg =
          case arg of
            True -> "Hello"
            False -> " case statements"

        main : List String -> String
        main args =
            (toText True) ++ (toText False)
        "#;
        let result = eval(src, None);
        insta::assert_snapshot!(result);
    }
}
