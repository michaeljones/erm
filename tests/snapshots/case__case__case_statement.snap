---
source: tests/case.rs
expression: result

---
Hello case statements
